<?xml version="1.0"?>
<launch>
  <arg name="sim" default="false"/>

  <include file="$(find bitbots_bringup)/launch/base.launch" >
    <arg name="sim" value="$(arg sim)"/>
  </include>

  <group unless="$(arg sim)">
    <include file="$(find bitbots_ros_control)/launch/ros_control.launch" />
  </group>

  <arg if="$(optenv IS_ROBOT false)" name="taskset" default="taskset -c 3"/>
  <arg unless="$(optenv IS_ROBOT false)" name="taskset" default=""/>

  <node name="dynamic_kick" pkg="bitbots_dynamic_kick" type="KickNode" output="screen" launch-prefix="$(arg taskset)">
    <remap from="/kick_motor_goals" to="/DynamixelController/command"/>
    <rosparam command="load" file="$(find bitbots_dynamic_kick)/config/kick_config.yaml" />
    <rosparam command="load" file="$(find bitbots_dynamic_kick)/config/kick_config_unstable.yaml" />
  </node>
</launch>
